Aptoide/appcoins-wallet-android
# [APPC-1102] Local Payments Integration

**What does this PR do?**

   Implemented Local Payment flow
   Reordered payment methods in order to have the available ones on top
   Refactored the way we are getting the bonus on the paymentMethods view in order to pass it to each of the payments fragments (in this ticket, only the local payments are receiving it)
  Upgraded Lottie’s lib
  Note: Translated Strings are still missing. Empty bonus string missing
  Added string for the payment methods that don’t give you bonus (ask a friend now gives bonus, so it should only appear in APPC)

**Database changed?**

No

**Where should the reviewer start?**

LocalPaymentFragment.kt
PaymentMethodsFragment.java
PaymentMethodsPresenter.java

**How should this be manually tested?**

  You can first check if the local payment are not appearing to you right now.
  Then check the comments on the ticket on how to make the local payments appear
  GoPay is the only one who supports dev/sandbox environment, so you must choose that one
  Here you have three flows to test:
  1. Click “Close page & return to merchant” and check if it shows the Pending User Payment screen
  2. Click “Proceed to GO-JEK app button and simulate a successful transaction, then go to Appcoins Wallet main view and check if the top up and bonus were made with the correct values (it may take a few seconds)
 3. Alter the code to always show the completed payment screen and after doing the flow in 1, check if the bonus value showed is the correct one
 4. Alter the code to first show the pending user payment animation and a few seconds later show the completed animation. Check if it works properly.
 In flow 1 and 3, the activity response should be canceled 
Test these steps in a OneStep Payment (buy oil in trivial drive)

Give a quick check trough some animations to check if the lib upgrade broke something
Switch between payment methods with and without to bonus and check if it behaves normally and if it shows the new string on the APPC method
 
**What are the relevant tickets?**

  https://aptoide.atlassian.net/browse/APPC-1102

**Code Review Checklist**
